### ISSUE TITLE
Open in a new tab by default

### ISSUE THREAD

**Issue by jamesray1** [MetaMask/metamask-extension#6111]
At the moment when I log into Metamask I would like to be able to autofill the password from LastPass. To do this I think Metamask needs to open in a new tab, rather than as a popup. Therefore I'd like to request that MetaMask opens in a new tab by default, or at least can be configured to do so.

**Comment by indigopsy** [comment]
It drives me crazy right now!!!

I need more than 10 mins to just sign a transaction.
Don't know why. don't care now.

But I can NOT click anywhere because MetaMask is a popup,
It disappears right away if I lose focus from popup.

Very very annoying

Please resolve it with:
Crtl + Left mouse click = new window
left mouse click only = popup

very eazy to implement

tnx :)

**Comment by foxreymann** [comment]
I suffer Repetitive strain injury (RSI) and can't use a mouse. I use web fully using my keyboard. But I can't do that with MetaMask as it opens in a pop up window with terrible accessibility. How to solve that? i would love the new tab option. It would work then with my Vimium chrome extension.

**Comment by bdresser** [comment]
hey all, just want to make sure you're aware of this option in the account menu. If you click the `...` you're able to open MetaMask in a new tab. Once it's open in a new tab, it will display all pending transaction or connect requests. They'll first show in the popup, but if you navigate to the tab, you'll see them there as well.

<img width="354" alt="Screen Shot 2019-05-15 at 9 48 37 AM" src="https://user-images.githubusercontent.com/1016190/57793735-e1923a80-76f6-11e9-93ef-0c06c10dda1b.png">

**Comment by jamesray1** [comment]
That doesn't show before logging in. I'd show a screenshot, but I can't in Firefox; when I try the popup disappears.

**Comment by jamesray1** [comment]
It's also problematic that MetaMask seems to keep forgetting my password. I copy and paste it via LastPass and it doesn't work, despite having not changed it, and it worked just yesterday. The UX is further worsened by needing to restore with a seed phrase and import my other account, every time this happens, which has been many times!

**Comment by bdresser** [comment]
@jamesray1 haven't heard the password issue before -- email support@metamask.io and we'll do our best to help out over there.

The point about allowing an option to enable full-tab by default is totally legit, will talk it over with the design team.

**Comment by sihoang** [comment]
> hey all, just want to make sure you're aware of this option in the account menu. If you click the `...` you're able to open MetaMask in a new tab. Once it's open in a new tab, it will display all pending transaction or connect requests. They'll first show in the popup, but if you navigate to the tab, you'll see them there as well.
> 
> <img alt="Screen Shot 2019-05-15 at 9 48 37 AM" width="354" src="https://user-images.githubusercontent.com/1016190/57793735-e1923a80-76f6-11e9-93ef-0c06c10dda1b.png">

@bdresser I was wondering if it's possible for dApps to trigger this setting instead of relying on users?

**Comment by jamesray1** [comment]
@sihoang, scroll up!

**Comment by sihoang** [comment]
@jamesray1 "or at least can be configured to do so."
You're asking from the users perspective.

I'm asking from the dapp developers perspective. This config may or may not be able to set by the dapp because it might make users confused when they intentionally set to "pop-up" mode and dapps just want to open in "tab" mode or vice-versa

**Comment by jamesray1** [comment]
> hey all, just want to make sure you're aware of this option in the account menu. If you click the ... you're able to open MetaMask in a new tab. Once it's open in a new tab, it will display all pending transaction or connect requests. They'll first show in the popup, but if you navigate to the tab, you'll see them there as well.

This has already been covered above.

https://github.com/MetaMask/metamask-extension/issues/6111#issuecomment-492734633

> That [the three dots] doesn't show before logging in. I'd show a screenshot, but I can't in Firefox; when I try the popup disappears.

**Comment by sihoang** [comment]
that requires the end users to set it.

I was trying to ask if it's possible for the dApp to set it so the users don't have to touch anything.  For example, when users click "Transact" on my dApp, can *my dApp* trigger Metamask to open up a tab instead of a pop up.

**Comment by bdresser** [comment]
@sihoang are you requesting this feature because a full tab would be easier to find than the popup or for a more specific reason?

**Comment by sihoang** [comment]
@bdresser yes, in my dapp, every time there're Metamask transaction, I display a dialog instructing users to look for the "pop up", but even then it's shown in our users study sessions that users still have a hard time finding the pop up, specially when they are doing screen share.

So it'd be nice if my dapp can set Metamask to always show as tab and our instructions specifically say look for the new tab.

**Comment by jamesray1** [comment]
This is feedback from Frank via Metamask Support:

> You can try copying and pasting the link below into your web browser (if you are using Chrome) to open MetaMask in a tab.
> chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/home.html
> 
> If you are using Firefox, you can use the link below.
> moz-extension://5696ad42-40c2-974b-acf8-9be6906c68cb/home.html
> 
I replied to say that the Firefox link doesn't work for me, while Metamask is enabled, but maybe it does for others?

**Comment by jamesray1** [comment]
OK, after logging in I am able to open it in a new tab. But that doesn't really solve the problem as I want to open it in a new tab before logging in, so that I can autofill the password in Lastpass.

**Comment by jamesray1** [comment]
Thanks to help from Frank with email support, here is how to open a new tab prior to logging in.

Log in to Metamask via the popup, if you aren't already logged in. You will only need to do this once. Then open it in a new tab: in the popup, click the Menu three dot icon, then click "Expand view". Save or bookmark the link. Use that in future. You can test by logging out and logging in again in the tab. You will still have to type or copy and paste the password via your password manager, as autofill with Lastpass is not supported. But copying and pasting into a page rather than a popup is easier.

Also, regarding apparently forgetting the password:

Frank:
> How often does MetaMask appear to forget your password? When it does seem like it does not take your password, have you tried typing it in manually? 
> 
> It appears that other LastPass users may have also experienced an issue with the "Copy Password" feature from LastPass where it does not copy the password and uses the old data that is saved on the clipboard. 
> https://forums.lastpass.com/viewtopic.php?f=12&t=308375&start=10
> 
> To test if you are having a similar problem as described in the LastPass forum above, I may recommend that you try pasting into a NotePad or Word document to see if LastPass did, in fact, copy the password for MetaMask. If it does not, then you may need to enter your password manually within MetaMask.
> 
> Again, I am very sorry if it is not currently feasible to open MetaMask in a new tab to autofill your password, but your feedback has been shared with our developers so that they can take it under consideration for future updates.

Me:
> If it happens again I'll try this and let you know, as well as copying and pasting. I think in the past I did copy and paste it into the URL bar of my browser and the password was as expected.

**Comment by ERC20s** [comment]
Dear all!
If you click 'Expand view' as shown above the save the URL as a bookmark!

This works for me

**Comment by owencraston** [comment]
It is not possible for extensions to inject content into each other. The only way this can be achieved is via the steps mentioned above. We have no plans at the moment to default to a tab view but we are working on introducing side panel view which _may_ alleviate some of these issues. If your issues persist, please open another issue.
